src/app/session/synced-session.service.spec.ts

```diff
@@ -0,0 +1,278 @@
+            expect(sessionService.getLoginState().getState()).toEqual(LoginState.loggedOut);
+            expect(sessionService.getSyncState().getState()).toEqual(SyncState.unsynced);
+            expect(sessionService.getConnectionState().getState()).toEqual(ConnectionState.disconnected);
+
+            expect(sessionService.isLoggedIn()).toEqual(false);
+            expect(sessionService.getCurrentUser()).not.toBeDefined();
+        });
+
+        it('has the correct state after Login with wrong credentials', async () => {
+            const loginState = await sessionService.login('demo', 'pass123');
+            expect(loginState).toEqual(LoginState.loginFailed);
+            expect(sessionService.getLoginState().getState()).toEqual(LoginState.loginFailed);
+            expect(sessionService.getSyncState().getState()).toEqual(SyncState.unsynced);
+
+            // remote session takes a bit longer than a local login - this throws on successful connection
+            await sessionService.getConnectionState().waitForChangeTo(ConnectionState.rejected, ConnectionState.connected);
+
+            expect(sessionService.isLoggedIn()).toEqual(false);
+            expect(sessionService.getCurrentUser()).not.toBeDefined();
+        });
+
+        it('has the correct state after Login with non-existing user', async () => {
+            const loginState = await sessionService.login('demo123', 'pass123');
+            expect(loginState).toEqual(LoginState.loginFailed);
+            expect(sessionService.getLoginState().getState()).toEqual(LoginState.loginFailed);
+            expect(sessionService.getSyncState().getState()).toEqual(SyncState.unsynced);
+
+            // remote session takes a bit longer than a local login - this throws on successful connection
+            await sessionService.getConnectionState().waitForChangeTo(ConnectionState.rejected, ConnectionState.connected);
+
+            expect(sessionService.isLoggedIn()).toEqual(false);
+            expect(sessionService.getCurrentUser()).not.toBeDefined();
+        });
+
+        it('has the correct state after Login with correct credentials', async () => {
+            const [loginState] = await Promise.all([
+                sessionService.login('demo', 'pass'),
+                sessionService.getSyncState().waitForChangeTo(SyncState.completed, SyncState.failed)
+            ]);
+            expect(loginState).toEqual(LoginState.loggedIn);
+            expect(sessionService.getLoginState().getState()).toEqual(LoginState.loggedIn);
+            expect(sessionService.getSyncState().getState()).toEqual(SyncState.completed);
+            expect(sessionService.getConnectionState().getState()).toEqual(ConnectionState.connected);
+
+            expect(sessionService.isLoggedIn()).toEqual(true);
+            expect(sessionService.getCurrentUser()).toBeDefined();
+        });
+
+        it('has the correct state after Logout', () => {
+            sessionService.logout();

```

Comment:
I would sugest to log the user in before calling logout. This way the test would also pass without even calling logout thus it does not properly test the function.